scripts-dev/mypy_synapse_plugin.py

```diff
@@ -151,21 +151,24 @@ def check_is_cacheable(signature: CallableType, ctx: MethodSigContext) -> None:
         ctx.api.fail(message, ctx.context, code=AT_CACHED_MUTABLE_RETURN)
 
 
-IMMUTABLE_BUILTIN_VALUE_TYPES = {
+IMMUTABLE_VALUE_TYPES = {
     "builtins.bool",
     "builtins.int",
     "builtins.float",
     "builtins.str",
     "builtins.bytes",
 }
 
-IMMUTABLE_BUILTIN_CONTAINER_TYPES = {
+IMMUTABLE_CONTAINER_TYPES_REQUIRING_HASHABLE_ELEMENTS = {
```

Comment:
I'm not suggesting one do this, but you could implement `__hash__` on something which is mutable. tl;dr I wonder if it makes sense to treat this different than `Sequence`?